tests/unit-tests/test_inferrer.py

```diff
@@ -391,7 +480,1004 @@ def check_eval(mock_do_mini_batch_inference,
+                            local_rank=0)
+        config = GSConfig(args)
+
+        model3 = create_builtin_edge_gnn_model(gdata.g, config, True)
+        inferrer3 = GSgnnEdgePredictionInferrer(model3)
+        inferrer3.setup_device(device)
+
+        infer_dataloader3 = GSgnnEdgeDataLoader(
+            gdata,
+            target_idx=gdata.get_edge_test_set(config.target_etype),
+            fanout=config.fanout,
+            batch_size=config.batch_size,
+            label_field=config.label_field,
+            node_feats=config.node_feat_name,
+            edge_feats=config.edge_feat_name,
+            remove_target_edge_type=config.remove_target_edge_type,
+            train_task=False)
+ 
+        with assert_raises(AssertionError):
+            inferrer3.infer(
+                loader=infer_dataloader3,
+                save_embed_path=None,
+                use_mini_batch_infer=False
+                )
+
+    with tempfile.TemporaryDirectory() as tmpdirname:
+        # get the test dummy distributed graph
+        _, part_config = generate_dummy_dist_graph(tmpdirname)
+        gdata = GSgnnData(part_config=part_config)
+
+        # Test case 3: abnormal case, set RGCN model without edge features for EC, but 
+        #              provide edge features.
+        #              This will trigger an assertion error, asking for projection weights
+        #              in the GSEdgeEncoderInputLayer.
+        create_config4ef(Path(tmpdirname), 'gnn_nc.yaml', task='ec', use_ef=False)
+        args = Namespace(yaml_config_file=os.path.join(Path(tmpdirname), 'gnn_nc.yaml'),
+                            local_rank=0)
+        config = GSConfig(args)
+
+        model4 = create_builtin_edge_gnn_model(gdata.g, config, True)
+        inferrer4 = GSgnnEdgePredictionInferrer(model4)
+        inferrer4.setup_device(device)
+
+        infer_dataloader4 = GSgnnEdgeDataLoader(
+            gdata,
+            target_idx=gdata.get_edge_test_set(config.target_etype),
+            fanout=config.fanout,
+            batch_size=config.batch_size,
+            label_field=config.label_field,
+            node_feats=config.node_feat_name,
+            edge_feats={('n0', 'r0', 'n1'): ['feat'],
+                        ('n0', 'r1', 'n1'): ['feat']},  # Manually set, as config does not have it.
+            remove_target_edge_type=config.remove_target_edge_type,
+            train_task=False)
+ 
+        with assert_raises(AssertionError):
+            inferrer4.infer(
+                loader=infer_dataloader4,
+                save_embed_path=None,
+                use_mini_batch_infer=True
+                )
+
+    with tempfile.TemporaryDirectory() as tmpdirname:
+        # get the test dummy distributed graph
+        _, part_config = generate_dummy_dist_graph(tmpdirname)
+        gdata = GSgnnData(part_config=part_config)
+
+        # Test case 4: abnormal case, set RGCN model with edge features for NC, but 
```

Comment:
```suggestion
        # Test case 4: abnormal case, set RGCN model with edge features for EC, but 
```

Please check other comments.

Reply:
changed